mpich-4.0b1 build error in 32bit mode..

Attempting a 32bit build on Fedora 35 with `gcc (GCC) 11.2.1 20210728 (Red Hat 11.2.1-1)`

```
../configure --prefix=$HOME/soft/mpich-4.0b1-32  CC=gcc CXX=g++ FC=gfortran F77=gfortran CFLAGS=-m32 CXXFLAGS=-m32 FCFLAGS='-m32 -fallow-argument-mismatch' FFLAGS='-m32 -fallow-argument-mismatch' --enable
-shared --with-device=ch3:sock --with-pm=hydra --enable-fast=no --enable-error-messages=all --enable-g
```
And get:
```
/usr/bin/ld: i386:x86-64 architecture of input file `/home/glci/soft/build/mpich-4.0b1/objdir/modules/hwloc/hwloc/.libs/libhwloc_embedded.a(topology.o)' is incompatible with i386 output
```
CFLAGS isn't getting passed down to hwloc?

[config.log](https://github.com/pmodels/mpich/files/7572805/config.log)
[make.log](https://github.com/pmodels/mpich/files/7572806/make.log)

Switching to the following gets the build going.
```
../configure --prefix=$HOME/soft/mpich-4.0b1-32  CC='gcc -m32'  CXX='g++ -m32' FC='gfortran -m32' F77='gfortran -m32' FCFLAGS='-fallow-argument-mismatch' FFLAGS='-fallow-argument-mismatch' --enable-shared
 --with-device=ch3:sock --with-pm=hydra --enable-fast=no --enable-error-messages=all --enable-g
```
But then:
```
[glci@pj03 f77]$ /home/glci/soft/mpich-4.0b1-32/bin/mpif90  fpi.f
/usr/bin/ld: /home/glci/soft/mpich-4.0b1-32/lib/libmpifort.so: undefined reference to `PMPI_Delete_error_code'
/usr/bin/ld: /home/glci/soft/mpich-4.0b1-32/lib/libmpifort.so: undefined reference to `PMPI_Delete_error_string'
/usr/bin/ld: /home/glci/soft/mpich-4.0b1-32/lib/libmpifort.so: undefined reference to `PMPI_Delete_error_class'
collect2: error: ld returned 1 exit status
```

> [glci@pj03 f77]$ /home/glci/soft/mpich-4.0b1-32/bin/mpif90  fpi.f
> /usr/bin/ld: /home/glci/soft/mpich-4.0b1-32/lib/libmpifort.so: undefined reference to `PMPI_Delete_error_code'
> /usr/bin/ld: /home/glci/soft/mpich-4.0b1-32/lib/libmpifort.so: undefined reference to `PMPI_Delete_error_string'
> /usr/bin/ld: /home/glci/soft/mpich-4.0b1-32/lib/libmpifort.so: undefined reference to `PMPI_Delete_error_class'
> collect2: error: ld returned 1 exit status

I am not reproducing this error. @balay Do you encounter similar error with 64-bit build?

--- a/confdb/aclocal_modules.m4
+++ b/confdb/aclocal_modules.m4
@@ -41,7 +41,7 @@ dnl internal routine, $1 is the extra cflags, hwloc_embedded_dir is m4 macro
 dnl defined to be the path to embedded hwloc.
 AC_DEFUN([PAC_CONFIG_HWLOC_EMBEDDED],[
     PAC_PUSH_FLAG([CFLAGS])
-    CFLAGS="$1"
+    CFLAGS="$USER_CFLAGS $1"
     hwloc_config_args="--enable-embedded-mode --disable-visibility"
     hwloc_config_args="$hwloc_config_args --disable-libxml2"
     hwloc_config_args="$hwloc_config_args --disable-nvml"

--- a/confdb/aclocal_subcfg.m4
+++ b/confdb/aclocal_subcfg.m4
@@ -1,24 +1,3 @@
-dnl PAC_RESET_ALL_FLAGS - Reset precious flags to those set by the user
-AC_DEFUN([PAC_RESET_ALL_FLAGS],[
-	if test "$FROM_MPICH" = "yes" ; then
-	   CFLAGS="$USER_CFLAGS"
-	   CPPFLAGS="$USER_CPPFLAGS"
-	   CXXFLAGS="$USER_CXXFLAGS"
-	   FFLAGS="$USER_FFLAGS"
-	   FCFLAGS="$USER_FCFLAGS"
-	   LDFLAGS="$USER_LDFLAGS"
-	   LIBS="$USER_LIBS"
-	fi
-])
-
-dnl PAC_RESET_LINK_FLAGS - Reset precious link flags to those set by the user
-AC_DEFUN([PAC_RESET_LINK_FLAGS],[
-	if test "$FROM_MPICH" = "yes" ; then
-	   LDFLAGS="$USER_LDFLAGS"
-	   LIBS="$USER_LIBS"
-	fi
-])
-
 dnl Sandbox configure with additional arguments
 dnl Usage: PAC_CONFIG_SUBDIR_ARGS(subdir,configure-args,action-if-success,action-if-failure)
 dnl

--- a/confdb/aclocal_util.m4
+++ b/confdb/aclocal_util.m4
@@ -58,6 +58,27 @@ AC_DEFUN([PAC_PREFIX_ALL_FLAGS],[
 	PAC_PREFIX_FLAG($1, EXTRA_LIBS)
 ])
 
+dnl PAC_RESET_ALL_FLAGS - Reset precious flags to those set by the user
+AC_DEFUN([PAC_RESET_ALL_FLAGS],[
+	if test "$FROM_MPICH" = "yes" ; then
+	   CFLAGS="$USER_CFLAGS"
+	   CPPFLAGS="$USER_CPPFLAGS"
+	   CXXFLAGS="$USER_CXXFLAGS"
+	   FFLAGS="$USER_FFLAGS"
+	   FCFLAGS="$USER_FCFLAGS"
+	   LDFLAGS="$USER_LDFLAGS"
+	   LIBS="$USER_LIBS"
+	fi
+])
+
+dnl PAC_RESET_LINK_FLAGS - Reset precious link flags to those set by the user
+AC_DEFUN([PAC_RESET_LINK_FLAGS],[
+	if test "$FROM_MPICH" = "yes" ; then
+	   LDFLAGS="$USER_LDFLAGS"
+	   LIBS="$USER_LIBS"
+	fi
+])
+
 AC_DEFUN([PAC_CHECK_FGREP_WORD],[
     AC_PROG_FGREP
     AC_MSG_CHECKING([if fgrep support -w option])

--- a/src/pm/hydra/configure.ac
+++ b/src/pm/hydra/configure.ac
@@ -11,6 +11,10 @@ AC_INIT([Hydra], MPICH_VERSION_m4)
 AC_CONFIG_AUX_DIR(confdb)
 AC_CONFIG_MACRO_DIR(confdb)
 
+if test "$FROM_MPICH" != "yes" ; then
+    PAC_PREFIX_ALL_FLAGS(USER)
+fi
+
 AC_ARG_PROGRAM
 
 dnl must come before LT_INIT, which AC_REQUIREs AC_PROG_CC

